### DISCUSSION TITLE
Authelia & Fail2Ban does not block ip

### DISCUSSION THREAD

**Discussion by Stogie87** [authelia/authelia#7931] [Q&A] [answered]
Hello everyone. I just can't solve the following problem: I want the corresponding IP address to be blocked for a certain period after 5 failed login attempts under Authelia. I am using Authelia in a Docker container as a host, and the Authelia logs also show that Authelia has banned the IP address and/or username. However, I can still make login attempts under Authelia without any issues using the same banned IP address, and as soon as I use the correct login credentials, Authelia lets me through normally and redirects me to the corresponding website. After several days of puzzling over this, I tried using the Fail2Ban container, which correctly reads the logs from Authelia and also correctly adds the corresponding IP address to the nftables within the Fail2Ban Docker and removes it after the ban period expires. Nevertheless, the problem remains that the banned IP address can still log in without any issues as soon as the correct user data is entered. I am using Unraid 7.0.0 beta2 and have installed Nginx Proxy Manager. I host my domain on Cloudflare, and Cloudflare correctly forwards domain requests to NPM, where I have then configured Authelia in between. I am very new to this area and have already tried to use ChatGPT for support over several days, but I'm just not making any progress here. I've attached my relevant configuration files and hope that someone can help me with this. Thank you in advance.

[Nginx Config für Nextcloud.txt](https://github.com/user-attachments/files/17162873/Nginx.Config.fur.Nextcloud.txt)
[Authelia configuration.yml.txt](https://github.com/user-attachments/files/17162874/Authelia.configuration.yml.txt)
[Auszug Fail2Ban Log.txt](https://github.com/user-attachments/files/17162876/Auszug.Fail2Ban.Log.txt)
[jail.d Authelia.local.txt](https://github.com/user-attachments/files/17162893/jail.d.Authelia.local.txt)
[General Jai.locall in Fail2Ban.txt](https://github.com/user-attachments/files/17162906/General.Jai.locall.in.Fail2Ban.txt)
![NFTTablesinFail2Ban png b28b0d28dcf34c36e765d969f624ce93](https://github.com/user-attachments/assets/e6e5d457-c21b-40df-9dcb-064cbce0fc4f)

**Comment by james-d-elliott** [comment]
Are you sure that is the full output of the iptables command? Seems either a lot has been omitted or is just missing.

**Reply by james-d-elliott** [reply]
Actually your issue is probably the IP. The IP iptables is able to see is the Cloudflare TCP IP address (because that's the proxy handling these requests), this is not useful for iptables as it'll block every request if it bans a Cloudflare IP, and will ban none if it bans the "real" IP since the requests come from the Cloudflare IP.

You need to instead configure the fail2ban action to generate a list of banned IP's and supply those to your proxy, and have your proxy block them.

**Reply by james-d-elliott** [reply]
https://vaarlion.com/blog/how-to-use-fail2ban-when-there-is-a-proxy-in-the-way/

**Reply by Stogie87** [reply]
> Are you sure that is the full output of the iptables command? Seems either a lot has been omitted or is just missing.

Yes, that is the full output i of iptables & nfttables i get inside the docker container of fail2ban.
Iptables in Unraid terminal shows this to me atm and it never shows any blocked ips by Fail2Ban or Authelia.
nfttables are not working at all atm in Unraid 7 for me. Idk why....
![grafik](https://github.com/user-attachments/assets/70939298-dae4-4272-998f-4ef6ec34ebb6)

**Reply by Stogie87** [reply]
> Actually your issue is probably the IP. The IP iptables is able to see is the Cloudflare TCP IP address (because that's the proxy handling these requests), this is not useful for iptables as it'll block every request if it bans a Cloudflare IP, and will ban none if it bans the "real" IP since the requests come from the Cloudflare IP.
> 
> You need to instead configure the fail2ban action to generate a list of banned IP's and supply those to your proxy, and have your proxy block them.

But i added this header in Nginx configuration according chatgpt^^:
    proxy_set_header X-Real-IP $remote_addr; 
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

**Reply by Stogie87** [reply]
> Actually your issue is probably the IP. The IP iptables is able to see is the Cloudflare TCP IP address (because that's the proxy handling these requests), this is not useful for iptables as it'll block every request if it bans a Cloudflare IP, and will ban none if it bans the "real" IP since the requests come from the Cloudflare IP.
> 
> You need to instead configure the fail2ban action to generate a list and supply those to your proxy, and have your proxy block them.

I also created a custom action in fail2ban which creates a txt.file (.config file is also possible) with the ip adresses, but idk how to make nginx proxy manager able to read this file and block the ips.

[custom bann action.txt](https://github.com/user-attachments/files/17163187/custom.bann.action.txt)

**Reply by Stogie87** [reply]
My Output-Files looks like this atm:
![grafik](https://github.com/user-attachments/assets/4fb0d20d-b4b1-4869-b021-3aac8a9b1617)

**Reply by james-d-elliott** [reply]
> I also created a custom action in fail2ban which creates a txt.file (.config file is also possible) with the ip adresses, but idk how to make nginx proxy manager able to read this file and block the ips.

Yeah that's probably the element you're missing. Think about it like this, iptables knows nothing about X-Forwarded-For, X-Real-IP, etc. It's purely the TCP/UDP src IP which is always Cloudflare given your architecture description. 

I'd suggest looking at the guide I linked, and if that doesn't make it clear then checking if any other NGINX Proxy Manager users have done similar, I suspect they probably have.

**Reply by nightah** [reply]
You can use the [Cloudflare fail2ban action](https://github.com/fail2ban/fail2ban/blob/master/config/action.d/cloudflare.conf).

It ensures that the IP in the X-Forwarded-For header is banned in Cloudflare for whatever period you set in your jail.

The same action will also manage unbanning and such.

**Reply by Stogie87** [reply]
> You can use the [Cloudflare fail2ban action](https://github.com/fail2ban/fail2ban/blob/master/config/action.d/cloudflare.conf).
> 
> It ensures that the IP in the X-Forwarded-For header is banned in Cloudflare for whatever period you set in your jail.
> 
> The same action will also manage unbanning and such.

Okay, i use cloudflare.conf action in addition now, but i see this output in the Fail2Ban-Log:
[Output.txt](https://github.com/user-attachments/files/17163474/Output.txt)

**Reply by Stogie87** [reply]
Fail2Ban iptables show me this now:
![grafik](https://github.com/user-attachments/assets/bdd62d2c-35dd-4fe2-9d7f-d9184c758335)

**Reply by Stogie87** [reply]
It looks like only nfttables work well inside the Fail2Ban Docker, if i activate my previous ban actions for allports and multiport:
![grafik](https://github.com/user-attachments/assets/ad43523e-4e5b-40ed-8f13-27c231163d8a)

**Reply by nightah** [reply, marked as answer]
When setup properly you will be able to see bans in the jail however nothing changes in iptables/nftables.

You can see the actual bans and the commentary in the Cloudflare dashboard. I can't remember the exact area but fairly certain it was called firewall or something like that.

**Reply by Stogie87** [reply]
> When setup properly you will be able to see bans in the jail however nothing changes in iptables/nftables.
> 
> You can see the actual bans and the commentary in the Cloudflare dashboard. I can't remember the exact area but fairly certain it was called firewall or something like that.

I think for that i would need cloudflare Pro?
I just use the free version.

**Comment by Stogie87** [comment]
Okay.
Thank you guys very much for your support @james-d-elliott & @nightah .
I could finally manage to block ips by using the cloudflare-token.conf as action with my cloudflare credentials.
I was literally working on this for the last 2 weeks and seems like i was using all the time the wrong action configs.
Now i can finally get my social life back! Thank you very much!

**Reply by james-d-elliott** [reply]
No worries glad it's all working! Cool solution from Amir too.
